packages/jest-circus/src/eventHandler.ts

```diff
@@ -147,7 +149,29 @@ const eventHandler: EventHandler = (event, state): void => {
       break;
     }
     case 'test_retry': {
+      // Clear errors so tests can be retried (and not immediately fail)
       event.test.errors = [];
+
+      // Clear any snapshot data that occurred in previous test run
+      const tmpSnapshotState = global.expect.getState().snapshotState;
+
+      if (tmpSnapshotState) {
+        if (
+          tmpSnapshotState._snapshotPath &&
```

Comment:
I took the freedom to start working on finishing this PR_(but let me know if you prefer me not to)_, I've made some good progress but one thing that is still not clear is: Why do we need to call `fs.unlinkSync(tmpSnapshotState._snapshotPath);`?

cc: @thymikee @palmerj3

Reply:
Hey! Please feel free to work on it :)  It's all of our code not my code.

Ultimately the issue this PR tries to fix is the fact that a snapshot file for a test will contain n-copies depending on how many times it's rerun. So this was one way we could fix that by removing the snapshot file before new data is written.

Reply:
Awesome! I'll keep working on it and update the PR once I have something 😄